src/server/main.cc

```diff
@@ -10,25 +10,42 @@
 #include <grpcpp/grpcpp.h>
 #include "src/proto/protocached.pb.h"
 #include "src/proto/protocached.grpc.pb.h"
+#include "src/server/data_store/data_store.h"
+#include "src/server/data_store/lru_cache_data_store.h"
 
 // TODO(joejoevictor): Move this to separate header and implementation file
 class ProtocachedServiceImpl final : public ::protocached::Protocached::Service {
+private:
+  DataStore* data_store;
+
+public:
+  ProtocachedServiceImpl() {
+    data_store = new LruCacheDataStore(100);
+  }
+
+  ~ProtocachedServiceImpl() {
+    delete data_store;
+  }
+
   ::grpc::Status Get(
     ::grpc::ServerContext* context,
     const ::protocached::GetRequest* request,
     ::protocached::GetResponse* response
   ) override {
     ::protocached::CachedValue cachedValue;
-    cachedValue.set_value("Received key: <" + request -> key() + ">" );
-    response -> mutable_value() -> CopyFrom(cachedValue);
+    ::std::string value_result = data_store -> Get(request -> key());
+    response -> mutable_value() -> set_value(value_result);
     return ::grpc::Status::OK;
-  } 
-  
+  }
+
   ::grpc::Status Set(
     ::grpc::ServerContext* context,
     const ::protocached::SetRequest* request,
     ::protocached::SetResponse* response
   ) override {
+    // TODO(joejoevictor): value().value() seems hacky...Is there a better way
```

Comment:
yeah, i feel like if our Set API takes (string key, CachedValue value), then it will avoid the value().value() part of it.